File: Homeworks/C# Part 2/06.StringsAndTextProcessing/03.CorrectBrackets/CorrectBrackets.cs
using System;

class CorrectBrackets
{
    static void Main()
    {
        string input = Console.ReadLine();
        int openCount = 0;
        int closeCount = 0;
        bool isCorrect = true;
        for (int i = 0; i < input.Length; i++)
        {
            if (input[i] == '(')
            {
                openCount++;
            }
            if (input[i] == ')')
            {
                closeCount++;
            }
            if (closeCount > openCount)
            {
                isCorrect = false;
                break;
            }
        }
        if (openCount != closeCount)
        {
            isCorrect = false;
        }
        Console.WriteLine(isCorrect ? "Correct" : "Incorrect");
    }
}

File: Homeworks/JS Fundamentals/11. Strings/02. Correct brackets/correct-brackets.js
function solve(args) {
	var input = args[0],
    	openCount = 0;
    	closeCount = 0;
    	isCorrect = true;
    for (var i = 0; i < input.length; i += 1) {
        if (input[i] === '(') {
            openCount += 1;
        }
        if (input[i] === ')') {
            closeCount += 1;
        }
        if (closeCount > openCount) {
            isCorrect = false;
            break;
        }
    }
    if (openCount != closeCount) {
        isCorrect = false;
    }
    console.log(isCorrect ? "Correct" : "Incorrect");
}